src/v2/components/Authentication.jsx

```diff
@@ -0,0 +1,110 @@
+ * @param {function} props.onAuthenticated - Callback to invoke when the user
+ *     authenticates successfully.
+ * @param {function} props.onCancel - Callback to invoke when the user dismisses
+ *     the authentication dialog.
+ * @param {string=} props.title - Optional title (defaults to `'Sign in'`).
+ *
+ * @returns {React.Element}
+ */
+export default function Authentication(props) {
+  const {onAuthenticated, onCancel, title = 'Sign in'} = props
+
+  const [username, setUsername] = useState('')
+  const [password, setPassword] = useState('')
+  const [message, setMessage] = useState('')
+
+  const handleCredentials = useCallback((event) => {
+    console.log('event')
+    event.preventDefault()
+    query
+      .post('/api/loginPassword', {username, password})
+      .then(onAuthenticated)
+      .catch(() => setMessage('Incorrect username or password'))
+  }, [onAuthenticated, password, username])
+
+  const handleFacebook = useCallback(() => {
+    const oauthUrl = getUrl(`/api/oauthFbUrl?path=%2Fauthenticated`)
+    const w = window.open(oauthUrl, 'c7e.authentication')
+    window.c7e = window.c7e || {}
+    window.c7e.authnCallback = () => {
+      onAuthenticated()
+      w.close()
+    }
+  }, [onAuthenticated])
+
+  const handleGoogle = useCallback(() => {
+    let oauthUrl
+    if (process.env.NODE_ENV === 'development') {
+      oauthUrl = getUrl(`/api/oauthDevUrl?email=secret@example.com&path=%2Fauthenticated`)
+    } else {
+      oauthUrl = getUrl(`/api/oauthUrl?path=%2Fauthenticated`)
+    }
+
+    const w = window.open(oauthUrl, 'c7e.authentication')
+    window.c7e = window.c7e || {}
+    window.c7e.authnCallback = () => {
+      onAuthenticated()
+      w.close()
+    }
+  }, [onAuthenticated])
+
+  const handleChange = useCallback(({target: {name, value}}) => {
+    switch (name) {
+      case 'username': return setUsername(value)
+      case 'password': return setPassword(value)
+      default:;
+    }
+  }, [])
+
+  return 0,
+  <Overlay onClick={onCancel}>
+    <div className="c7e-authentication">
+      <h2 className="c7e-authentication--title">{title}</h2>
+      <div className="c7e-authentication--oauth">
+        <Button className="c7e-authentication--button" icon="c7e-logomark--google-g" label="Sign in with Google" onClick={handleGoogle}/>
+        <Button className="c7e-authentication--button" icon="c7e-logomark--facebook-f" label="Sign in with Facebook" onClick={handleFacebook}/>
+      </div>
+      <div className="c7e-authentication--divider">
+        <div className="c7e-authentication--divider-label">or</div>
+      </div>
+      <form className="c7e-authentication--credentials" onSubmit={handleCredentials}>
+        { message && <div className="c7e-authentication--message">{message}</div> }
+        <TextInput placeholder="Username" onChange={handleChange}/>
+        <TextInput type="password" placeholder="Password" onChange={handleChange}/>
```

Comment:
I added the `autocomplete` and `id` attributes.

`required` doesn’t do anything since this is a React form and we’re handling everything (i.e., we’re required to handle everything by React). If we want to do form validation, even just to require fields, that’s on us.